### PR TITLE
!typ Rename ActorContext#upcast to ActorContext#unsafeUpcast.

### PR DESCRIPTION
**Pull request by He-Pin**
Refs:https://github.com/akka/akka/issues/25896

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala [hunk 1]
```diff
@@ -37,9 +37,10 @@ trait ActorRef[-T] extends RecipientRef[T] with java.lang.Comparable[ActorRef[_]
   /**
    * Unsafe utility method for widening the type accepted by this ActorRef;
    * provided to avoid having to use `asInstanceOf` on the full reference type,
-   * which would unfortunately also work on non-ActorRefs.
+   * which would unfortunately also work on non-ActorRefs.Use it with caution,it may cause a [[ClassCastException]] when you send a message
+   * to the widened [[ActorRef[U]]].
```
Comment (hepin1989) [inline akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: Feel safer.

**Comment by hepin1989** [comment]
@rkuhn @patriknw I have updated the PR with some additional scaladoc.

**Review by chbatey — APPROVED** [review]
Good rename IMO. And glad we didn't use it very often!

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala (hunk 1 above)
Comment (hepin1989) [reply akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala]: @chbatey Done, thanks for review.

**Comment by hepin1989** [comment]
Wow, got merged, love it.

**Comment by dwijnand** [comment]
Nicely done @hepin1989!

**Comment by dwijnand** [comment]
@chbatey 

> And glad we didn't use it very often!

Makes you think right?  Unfortunately, those are only the ones that use the utility method.  Are there others that just cast?

**Pull request merged**